### PR TITLE
Enforce declaration order

### PR DESCRIPTION
**Pull request by npepinpe**
## Description

Adds a new Checkstyle check to enforce declaration order. The main difference is that fields are now only grouped by static/instance, and the final modifier is ignored. So before, we had:

```java
public class MyClass {
  private static final int A = 1;
  public static int b;

  private final String topic;
  protected boolean isRunning;
  // ...
}
```

Which now becomes:

```java
public class MyClass {
  public static int b;
  private static final int A = 1;

  protected boolean isRunning;
  private final String topic;

 // ...
}
```

I think the trade off here that it's now enforced is worth it.

## Pull Request Checklist

- [x] All commit messages match our [commit message guidelines](https://github.com/zeebe-io/zeebe/blob/develop/CONTRIBUTING.md#commit-message-guidelines)
- [x] The submitting code follows our [code style](https://github.com/zeebe-io/zeebe/wiki/Code-Style)
- [x] If submitting code, please run `mvn clean install -DskipTests` locally before committing

### CODE REVIEW

**Comment by npepinpe** [comment]
I've split it in two commits; first one is the update to the build tools, second one is reformatting.

One thing to note with the IntelliJ formatter though, it doesn't know about forward declarations :disappointed: So if you ask it to rearrange stuff it will blindly rearrange it without care. Checkstyle is aware of forward declarations though so it's just a matter of manually reordering (if you use the IntelliJ formatter).

File: build-tools/src/main/resources/intellij/ZeebeStyle.xml [hunk 1]
```diff
@@ -115,54 +118,6 @@
...
-              <AND>
-                <FIELD>true</FIELD>
-                <FINAL>true</FINAL>
-                <PROTECTED>true</PROTECTED>
-                <STATIC>true</STATIC>
-              </AND>
-            </match>
-          </rule>
-        </section>
-        <section>
```
Comment (Zelldon) [inline build-tools/src/main/resources/intellij/ZeebeStyle.xml]: Why are these removed?

Comment (npepinpe) [reply build-tools/src/main/resources/intellij/ZeebeStyle.xml]: I removed the rules that enforced grouping final fields (static and instance) together, since the check would flag them as wrong (since it'd order final private above non-final public, for example).

Comment (Zelldon) [reply build-tools/src/main/resources/intellij/ZeebeStyle.xml]: ok thanks

**Review by ChrisKujawa — APPROVED** [review]
